napari/components/_layers_list/view/main.py

```diff
@@ -1,18 +1,143 @@
+                index = self.vbox_layout.indexOf(layer._qt_properties)
+                divider = self.vbox_layout.itemAt(index+1).widget()
+                self.vbox_layout.removeWidget(layer._qt_properties)
+                self.vbox_layout.removeWidget(divider)
+                self.vbox_layout.insertWidget(2*(total - i)-1,
+                                              layer._qt_properties)
+                self.vbox_layout.insertWidget(2*(total - i), divider)
+
+    def mouseReleaseEvent(self, event):
+        """Unselects all layer widgets
+        """
+        self.layers.unselect_all()
+
+    def dragLeaveEvent(self, event):
+        """Unselects layer dividers
+        """
+        event.ignore()
+        for i in range(0, self.vbox_layout.count(), 2):
+            self.vbox_layout.itemAt(i).widget().setSelected(False)
+
+    def dragEnterEvent(self, event):
+        event.accept()
+        divs = []
+        for i in range(0, self.vbox_layout.count(), 2):
+            widget = self.vbox_layout.itemAt(i).widget()
+            divs.append(widget.y()+widget.frameGeometry().height()/2)
+        self.centers = [(divs[i+1]+divs[i])/2 for i in range(len(divs)-1)]
+
+    def dragMoveEvent(self, event):
+        cord = event.pos().y()
+        center_list = (i for i, x in enumerate(self.centers) if x > cord)
+        divider_index = next(center_list, len(self.centers))
+        layerWidget = event.source()
+        total = self.vbox_layout.count()//2 - 1
+        index = total - self.vbox_layout.indexOf(layerWidget)//2 - 1
+        insert = total - divider_index
+        if not (insert == index) and not (insert-1 == index):
+            state = True
+        else:
+            state = False
+        for i in range(0, self.vbox_layout.count(), 2):
+            if i == 2*divider_index:
+                self.vbox_layout.itemAt(i).widget().setSelected(state)
+            else:
+                self.vbox_layout.itemAt(i).widget().setSelected(False)
+
+    def dropEvent(self, event):
+        for i in range(0, self.vbox_layout.count(), 2):
+            self.vbox_layout.itemAt(i).widget().setSelected(False)
+        cord = event.pos().y()
+        center_list = (i for i, x in enumerate(self.centers) if x > cord)
+        divider_index = next(center_list, len(self.centers))
+        layerWidget = event.source()
+        total = self.vbox_layout.count()//2 - 1
+        index = total - self.vbox_layout.indexOf(layerWidget)//2 - 1
+        insert = total - divider_index
+        if index != insert and index+1 != insert:
+            if not self.layers[index].selected:
+                self.layers.unselect_all()
+                self.layers[index].selected = True
+            self.layers._move_layers(index, insert)
+        event.accept()
+
+
+class QtDivider(QFrame):
+    unselectedStlyesheet = "QFrame {;}"
+    selectedStlyesheet = """QFrame {border: 2px solid %s;
+        background-color: white; border-radius: 3px;}""" % palette['text']
```

Comment:
This seems like some pretty stringent hardcoding... Can this be moved to our styles resources?

Reply:
this is a question for @freeman-lab. I asked him the other day and he said no and that there was something to do with how we were doing our interactive updating of the element. I'm not sure if he's looked into this again. But I can check it out in a follow up PR as I agree - it should not live here

Reply:
Looks like I overlooked this in the initial refactor. It should follow the same technique as described in #167 using properties (i.e. setting whether it's selected or not).